src/porepy/params/bc.py

```diff
@@ -59,13 +57,6 @@ class BoundaryCondition(AbstractBoundaryCondition):
             face i has been assigned a Neumann condition.
         is_rob (np.ndarray, boolean, size g.num_faces): Element i is true if
             face i has been assigned a Robin condition.
-        is_per (np.ndarray, boolean, size g.num_faces): Element i is true if
```

Comment:
This is perhaps conceptually important: You've moved periodicity from a boundary condition concept to a property of the grid. The only practical difference that I can see is if we implement periodicity for more discretizations, there are fewer places we need to set periodicity (g only, instead of multiple bcs in a multi-physics problem). 

Have you thought deeper about this?